src/controllers/TalksController.php

```diff
@@ -498,12 +498,89 @@ protected function getTalkDataFromRequest(PDO $db, Request $request, $event_id)
             FILTER_SANITIZE_URL
         );
 
-        $talk['speakers'] = array_map(function ($speaker) {
-            $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
-            $speaker = trim($speaker);
-            return $speaker;
-        }, (array) $request->getParameter('speakers'));
+        $talk['speakers'] = array_map(
+            function ($speaker) {
+                $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
+                $speaker = trim($speaker);
+                return $speaker;
+            },
+            (array) $request->getParameter('speakers')
+        );
+
+        return $talk;
+    }
+
+    public function getSpeakersForTalk(Request $request, PDO $db)
+    {
+        $talk_id = $this->getItemId($request);
+        $talk = $this->getTalkById($db, $request, $talk_id);
+        $collection = new TalkModelCollection([$talk], 1);
+        return $collection->getTalks()[0]->speakers;
+    }
+
+    public function linkUserToTalk(Request $request, PDO $db)
+    {
+        //We must be logged in
+        if (!isset($request->user_id)) {
+            throw new Exception("You must be logged in to create data", 401);
+        }
+
+        $talk_id = $this->getItemId($request);
+        $talk_mapper = new TalkMapper($db, $request);
+        $talk = $talk_mapper->getTalkById($talk_id);
+        if (! $talk) {
+            throw new Exception("Talk not found", 404);
+        }
+
+        $user_id = $request->user_id;
+        $user_mapper = new UserMapper($db, $request);
+        $user = $user_mapper->getUserById($user_id)['users'][0];
+
+        $data = $this->getLinkUserDataFromRequest($request);
+
+        if ($data['display_name'] == '' || $data['username'] == '') {
+            throw new Exception("You must provide a display name and a username", 400);
+        }
+
+        //Get the speaker record based on the display name - check if this is already claimed,
+        //otherwise ID becomes the claim_id
+
+        $claim = $talk_mapper->getSpeakerFromTalk($talk_id, $data['display_name']);
+
+        if (! $claim) {
+            throw new Exception("No speaker matching that name found", 422);
+        }
+
+        if ($claim && $claim['speaker_id'] != null) {
+            throw new Exception("Talk already claimed", 422);
+        }
 
+        $pending_talk_claim_mapper = new PendingTalkClaimMapper($db, $request);
+
+        //Is the speaker this user?
+        if ($data['username'] === $user['username']) {
```

Comment:
I didn't test this, but what happens when the current user is the event-admin and wants to claim his talk on the event? Happens sometimes in our usergroup. I know it's an edge case, but it wouldn't be possible to approve my own claim for a talk with this setup, wouldn't it?